Document the possibility of cloud hosted software (#58)
DanNixon/sine2020_software_guidelines_paper: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex

It is recommended to limit the versions of particular operating systems you support.
Typically you should stop supporting an OS version when the publisher stops supporting it.

> An increasingly common method of delivering software to users is via cloud infrastructure.

Reviewer: ```suggestion
Cloud infrastructure is becoming an increasingly common method for delivering software to users. 
```

Author: What about "The cloud" rather than "Cloud infrastructure"?

Reviewer: cloud computing?

Author: :+1:

> In this model the software is designed to run on dedicated compute infrastructure with an interface layer providing interaction with the user.

Reviewer: ```suggestion
Instead of the user providing the computing hardware, an interface layer (e.g. web interface, command line) is provided to an external dedicated computing infrastructure. 
```

Author: Remove "external", this could be public or facility hosted cloud.

Reviewer: I meant external as in not your local machine.

Author: I don't think that's obvious here.

> This layer may be a desktop based GUI, command line or web interface.

Reviewer: ```suggestion
```

> The advantage of this delivery method is that the environment is which the software runs is more closely controlled and specified by the developer, reducing the impact of many cross platform development issues.

Reviewer: ```suggestion
One of the main advantages of cloud infrastructure is that the exact specifications of the operating system and hardware are known by the developer.
This makes it easier to recreate platform specific bugs and if cloud infrastructure is used exclusively it could remove the need for cross platform development.
```

Author: Maybe remove "exact".
You don't always know what you get, especially on public cloud or badly managed facility cloud.

(so pretty much everywhere I guess...)

> The software may also reside closer to the data, as data volume grows this is important in removing data transfer and storage bottlenecks involved in traditional software.

Reviewer: ```suggestion
It is also possible to create a setup where the data is easily accessible from the cloud infrastructure. 
This removes the need for data transfer and storage bottlenecks, which will become increasingly important as data volumes grow.
```

> A disadvantage of this method is that it add dependency on the infrastructure hosting the software.

Reviewer: ```suggestion
However, a disadvantage of cloud infrastructure is that it creates an additional dependency.
If the cloud infrastructure is not maintained correctly than the software will become unusable within the cloud service.
```

Reviewer: not sure on the second sentence

Author: Yeah, that's not really my original point.
I was more hinting at the fact you are not dependent on someone else's infrastructure which must be maintained and paid for and hope that it doesn't disappear overnight or significantly change.
The software is less of a concern as that is for the developer to maintain.

Reviewer: could we say something more along the lines of your comment?
